File: Logic Building Problems/Easy Problems/Program for Sum of Digits of a Number/method2.js
// Recursive JavaScript Code to find sum of digits

function sumOfDigits(n) {
    
    // Base Case
    if (n == 0)
        return 0;

    // Recursive Case
    return (n % 10) + sumOfDigits(Math.floor(n / 10));
}

// Driver code
console.log(sumOfDigits(12345));

File: Logic Building Problems/Easy Problems/Program for Sum of Digits of a Number/method2.java
// Recursive Java Code to find sum of digits

import java.io.*;

class Main {
    static int sumOfDigits(int n) {
        
        // Base Case
        if (n == 0)
            return 0;

        // Recursive Case
        return (n % 10) + sumOfDigits(n / 10);
    }

    public static void main(String[] args) {
        System.out.println(sumOfDigits(12345));
    }
}